File: API/_archive/REST/golang/usecase5_patch_updates.go
package main

import (
	"encoding/json"
	"log"
	"net/http"
)

func patchUserHandler(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPatch {
		http.Error(w, "Use PATCH", http.StatusMethodNotAllowed)
		return
	}
	
	var updates map[string]interface{}
	json.NewDecoder(r.Body).Decode(&updates)
	
	// Apply only the fields present in the 'updates' map to the DB
	log.Printf("Updating fields: %v", updates)
	
	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(map[string]string{"status": "updated"})
}

func main() {
	http.HandleFunc("/users/1", patchUserHandler)
	http.ListenAndServe(":8080", nil)
}

File: API/_archive/REST/python/usecase5_patch_updates.py
from fastapi import FastAPI, Body
import uvicorn

app = FastAPI()

@app.patch("/users/{user_id}")
async def update_user(user_id: str, payload: dict = Body(...)):
    # payload will only contain fields the client explicitly sent
    print(f"Applying updates to {user_id}: {payload}")
    return {"status": "updated", "data": payload}

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8000)
